wolfcrypt/src/port/Espressif/esp_crt_bundle/esp_crt_bundle.c

```diff
@@ -0,0 +1,1380 @@
+#if defined(WOLFSSL_DEBUG_CERT_BUNDLE) && defined(WOLFSSL_DEBUG_IGNORE_ASN_TIME)
+    esp_show_current_datetime();
+
+    if ((preverify == 0) && (store->error == ASN_BEFORE_DATE_E)) {
+        preverify = wolfssl_ssl_conf_verify_cb_before_date(preverify, store);
+    }
+
+    if ((preverify == 0) && (store->error == ASN_AFTER_DATE_E)) {
+        preverify = wolfssl_ssl_conf_verify_cb_after_date(preverify, store);
+    }
+#endif
+
+    /* Insert any other callback handlers here. */
+
+#ifdef WOLFSSL_DEBUG_CERT_BUNDLE
+    /* When debugging, show if have we resolved any error. */
+    if (preverify == 1) {
+        ESP_LOGCBI(TAG, "Returning preverify == 1\n");
+        if (preverify != initial_preverify) {
+            /* Here we assume wolfssl_ssl_conf_verify_cb_no_signer
+             * properly found and validated the problem: such as
+             * a new cert from the bundled needed for signing. */
+            ESP_LOGCBW(TAG, "Callback overriding error initial preverify = %d, "
+                            "returning preverify = %d",
+                            initial_preverify, preverify );
+        }
+    }
+    else {
+        ESP_LOGCBW(TAG, "Warning; returning preverify == %d\n", preverify);
+    }
+#endif
+
+    return preverify;
+} /* wolfssl_ssl_conf_verify_cb */
+
+/* wolfssl_ssl_conf_verify() patterned after ESP-IDF.
+ * Used locally here only. Not used directly by esp-tls.
+ *
+ * This is typically called during esp_crt_bundle_attach() in
+ * *this* file, which has same-name functions gated with the macro:
+ *   CONFIG_ESP_TLS_USING_WOLFSSL
+ *
+ * See also ESP-IDF transport_ssl component. */
+void CB_INLINE wolfssl_ssl_conf_verify(wolfssl_ssl_config *conf,
+                             int (*f_vrfy) WOLFSSL_X509_VERIFY_CALLBACK,
+                             void (*p_vrfy) )
+{
+    /* Other Crytographic providers for reference:
+    conf->f_vrfy      = f_vrfy;  (verification function callback)
+    conf->p_vrfy      = p_vrfy;  (pre-verification value)
+    */
+
+    /* typedef int (*VerifyCallback)(int, WOLFSSL_X509_STORE_CTX*); */
+    wolfSSL_CTX_set_verify( (WOLFSSL_CTX *)(conf->priv_ctx),
+                            WOLFSSL_VERIFY_PEER, wolfssl_ssl_conf_verify_cb);
+}
+
+/* esp_crt_verify_callback() patterned after ESP-IDF.
+ * Used locally here only. Not used directly by esp-tls.
+ *
+ * This callback is called for every certificate in the chain. If the chain
+ * is proper each intermediate certificate is validated through its parent
+ * in the x509_crt_verify_chain() function. So this callback should
+ * only verify the first untrusted link in the chain is signed by the
+ * root certificate in the trusted bundle
+*/
+int esp_crt_verify_callback(void *buf, WOLFSSL_X509 *crt, int depth,
+                            uint32_t *flags)
+{
+    WOLFSSL_X509 *child;
+    const uint8_t *crt_name;
+    int ret = -1;
+    int start = 0;
+    int end  = 0;
+    int middle =0;
```

Comment:
Nit: Space after =

Reply:
Yes, always appreciated, fixed. I also updated the order.